frontend/src/metabase/dashboard/dashboard.js

```diff
@@ -188,6 +188,11 @@ export const addCardToDashboard = function({
     };
     dispatch(createAction(ADD_CARD_TO_DASH)(dashcard));
     dispatch(fetchCardData(card, dashcard, { reload: true, clear: true }));
+
+    // guard in case card was filtered
+    if (card.dataset_query && card.dataset_query.database) {
```

Comment:
I'm not 100% sure this guard is necessary. Similar logic existed in `fetchCardData`, so I copied it.